gwcs/extension.py

```diff
@@ -1,34 +1,64 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-import os
-from asdf import util
-from asdf.extension import BuiltinExtension
-from .tags.wcs import *  # noqa
-from .tags.selectortags import *  # noqa
-from .tags.spectroscopy_models import * # noqa
-from .tags.geometry_models import * # noqa
+from asdf.extension import ManifestExtension
+from .converters.wcs import (
+    CelestialFrameConverter, CompositeFrameConverter, FrameConverter,
+    Frame2DConverter, SpectralFrameConverter, StepConverter,
+    StokesFrameConverter, TemporalFrameConverter, WCSConverter,
+)
+from .converters.selector import (
+    LabelMapperConverter, RegionsSelectorConverter
+)
+from .converters.spectroscopy import (
+    GratingEquationConverter, SellmeierGlassConverter, SellmeierZemaxConverter,
+    Snell3DConverter
+)
+from .converters.geometry import (
+    DirectionCosinesConverter, SphericalCartesianConverter
+)
 
-# Make sure that all tag implementations are imported by the time we create
-# the extension class so that _gwcs_types is populated correctly.
-from .tags import *  # noqa
-from .gwcs_types import _gwcs_types
 
+WCS_MODEL_CONVERTERS = [
+    CelestialFrameConverter(),
+    CompositeFrameConverter(),
+    FrameConverter(),
+    Frame2DConverter(),
+    SpectralFrameConverter(),
+    StepConverter(),
+    StokesFrameConverter(),
+    TemporalFrameConverter(),
+    WCSConverter(),
+    LabelMapperConverter(),
+    RegionsSelectorConverter(),
+    GratingEquationConverter(),
+    SellmeierGlassConverter(),
+    SellmeierZemaxConverter(),
+    Snell3DConverter(),
+    DirectionCosinesConverter(),
+    SphericalCartesianConverter(),
+]
 
-SCHEMA_PATH = os.path.abspath(
-    os.path.join(os.path.dirname(__file__), 'schemas'))
+# The order here is important; asdf will prefer to use extensions
+# that occur earlier in the list.
+WCS_MANIFEST_URIS = [
+    "asdf://stsci.edu/schemas/gwcs/manifests/gwcs-schemas-1.0",
+]
 
 
-class GWCSExtension(BuiltinExtension):
-    @property
-    def types(self):
-        return _gwcs_types
+TRANSFORM_EXTENSIONS = [
+    ManifestExtension.from_uri(
```

Comment:
Probably I got inspired from https://github.com/spacetelescope/roman_datamodels/blob/3df042878a2981190a1e2eb4c985cc4a4d64eec3/src/roman_datamodels/extensions.py

But it also allows easily extending things. And I do not think this would have _any measurable_ performance penalty.

Reply:
You are correct, though I wonder if that will ever happen.

Reply:
Unfortunately for you, I possess the world's tiniest stopwatch.

_measured_